src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs

```diff
@@ -193,10 +236,13 @@ public static bool IsLockFileStillValid(DependencyGraphSpec dgSpec, PackagesLock
 
                                     if (p2pSpecProjectRestoreMetadataFrameworkInfo != null)
                                     {
-                                        if (HasP2PDependencyChanged(p2pSpecTargetFrameworkInformation.Dependencies, p2pSpecProjectRestoreMetadataFrameworkInfo.ProjectReferences, projectDependency, dgSpec))
+                                        (var hasChanged, var message) = HasP2PDependencyChanged(p2pSpecTargetFrameworkInformation.Dependencies, p2pSpecProjectRestoreMetadataFrameworkInfo.ProjectReferences, projectDependency, dgSpec);
+
+                                        if (hasChanged)
                                         {
                                             // P2P transitive package dependencies have changed
-                                            return false;
+                                            // TODO NK - stuck here.
```

Comment:
todo means the PR is not ready, or needs a link to a tracking issue.

Reply:
removed TODO as it seems be unnecessary here because P2P dependencies having changed invalidating lock file.

Reply:
I think the stuck here was added in the code because not all scenarios were considered.


Unsure if that's the case, not doing a super in depth review now given that you're tweaking things per Andy's feedback.